discovery/azure/azure.go

```diff
@@ -508,3 +511,16 @@ func (client *azureClient) getNetworkInterfaceByID(networkInterfaceID string) (n
 
 	return result, nil
 }
+
+func getPowerStateFromVMInstanceView(instanceView *compute.VirtualMachineInstanceView) (powerState string) {
```

Comment:
While rebasing my PR #4590 I found this function to always panic Prometheus for me with

```
panic: runtime error: invalid memory address or nil pointer dereference                                    
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x14fd62e]
```

It seems that `instanceView.Statuses == nil` is the culprit here, because when checking instead `instanceView == nil` it works.

Reply:
This happens not only in my PR branch, but also in master for me :(

Reply:
Hi @marratj Thanks for reporting this. I am surprised that we have a case of nil instanceView :( . That is not to happen. Could you please email me at tariq.ibrahim@microsoft.com , so that I can investigate that further?

I will , in the mean time, work on adding the nil check here.

Reply:
PR #4977 created